src/vs/workbench/parts/debug/common/debugModel.ts

```diff
@@ -490,6 +490,18 @@ export class Thread implements IThread {
 		});
 	}
 
+	/**
+	 * Returns exception info promise if the exception was thrown and the debug adapter supports 'exceptionInfo' request, otherwise null
+	 */
+	public get exceptionInfo(): TPromise<DebugProtocol.ExceptionInfoResponse> {
+		const session = this.process.session;
```

Comment:
I would love if the exception widget always calls this method.
And all the logic is in this method, meaning that if we are not stopped we return null, if the session does not supportExceptionInfoRequest we also return something